<a href="https://colab.research.google.com/github/irwanmaulana42/google-collab/blob/master/Analisis_Penjualan_Supermarket%20with%20Bokeh.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Pendahuluan

Import Library

In [2]:
import pandas as pd
import numpy as np

from bokeh.io import show, output_notebook
from bokeh.models import ColumnDataSource, FactorRange, DatetimeTickFormatter
from bokeh.plotting import figure
from bokeh.transform import factor_cmap
from bokeh.models import NumeralTickFormatter
from bokeh.palettes import Spectral3,Spectral4,Spectral5
from bokeh.transform import cumsum
from math import pi


Membaca data yang dimiliki

In [3]:
df=pd.read_csv('https://raw.githubusercontent.com/irwanmaulana42/google-collab/master/supermarket_sales.csv')
df.head()

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   
4  373-73-7910      A     Yangon        Normal    Male   

             Product line  Unit price  Quantity   Tax 5%     Total       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715   1/5/2019   
1  Electronic accessories       15.28         5   3.8200   80.2200   3/8/2019   
2      Home and lifestyle       46.33         7  16.2155  340.5255   3/3/2019   
3       Health and beauty       58.22         8  23.2880  489.0480  1/27/2019   
4       Sports and travel       86.31         7  30.2085  634.3785   2/8/2019   

    Time      Payment    cogs  gross margin percentage  gross income  Rating  
0  13:08      Ewallet  522.83                 4.761905       26.1415     9.1  
1  10:29         Cash   76.40                 4.761905        3.8200     9.6  
2  13:23  Credit card  324.31                 4.761905       16.2155     7.4  
3  20:33      Ewallet  465.76                 4.761905       23.2880     8.4  
4  10:37      Ewallet  604.17                 4.761905       30.2085     5.3

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Invoice ID               1000 non-null   object 
 1   Branch                   1000 non-null   object 
 2   City                     1000 non-null   object 
 3   Customer type            1000 non-null   object 
 4   Gender                   1000 non-null   object 
 5   Product line             1000 non-null   object 
 6   Unit price               1000 non-null   float64
 7   Quantity                 1000 non-null   int64  
 8   Tax 5%                   1000 non-null   float64
 9   Total                    1000 non-null   float64
 10  Date                     1000 non-null   object 
 11  Time                     1000 non-null   object 
 12  Payment                  1000 non-null   object 
 13  cogs                     1000 non-null   float64
 14  gross margin percentage  

Menambahkan kolom 'Bulan' pada tabel

In [ ]:
df['Date'] = pd.to_datetime(df.Date)
df['Bulan'] = pd.to_datetime(df['Date']).dt.month
df['Tahun'] = pd.to_datetime(df['Date']).dt.year
df.head()

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   
4  373-73-7910      A     Yangon        Normal    Male   

             Product line  Unit price  Quantity   Tax 5%     Total       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715 2019-01-05   
1  Electronic accessories       15.28         5   3.8200   80.2200 2019-03-08   
2      Home and lifestyle       46.33         7  16.2155  340.5255 2019-03-03   
3       Health and beauty       58.22         8  23.2880  489.0480 2019-01-27   
4       Sports and travel       86.31         7  30.2085  634.3785 2019-02-08   

    Time      Payment    cogs  gross margin percentage  gross income  Rating  \
0  13:08      Ewallet  522.83                 4.761905       26.1415     9.1   
1  10:29         Cash   76.40                 4.761905        3.8200     9.6   
2  13:23  Credit card  324.31                 4.761905       16.2155     7.4   
3  20:33      Ewallet  465.76                 4.761905       23.2880     8.4   
4  10:37      Ewallet  604.17                 4.761905       30.2085     5.3   

   Bulan  Tahun  
0      1   2019  
1      3   2019  
2      3   2019  
3      1   2019  
4      2   2019

Isi

Pertanyaan 1: Pada bulan apa penjualan terbanyak?


In [ ]:
df['Bulan'].value_counts()

1    352
3    345
2    303
Name: Bulan, dtype: int64

In [ ]:
hasil = df.groupby('Bulan').sum()

In [ ]:
output_notebook()

# Convert 'bulan' to string representation
bulan = [str(Bulan) for Bulan, df in df.groupby('Bulan')]

source = ColumnDataSource(data=dict(bulan=bulan, total=hasil['Total']))

p = figure(x_range=FactorRange(*bulan), height=350, title="Total Penjualan per Bulan",
           toolbar_location=None, tools="")


p.vbar(x='bulan', top='total', width=0.9, source=source, color="steelblue")


p.y_range.start = 0
p.xgrid.grid_line_color = None
p.xaxis.axis_label = "Bulan"
p.yaxis.axis_label = "Total dalam USD ($)"
p.xaxis.major_label_orientation = 1.2
p.yaxis.formatter = NumeralTickFormatter(format="0")

show(p)

In [ ]:
hasil['Total']

Bulan
1    116291.868
2     97219.374
3    109455.507
Name: Total, dtype: float64

Berdasarkan hasil diatas dapat dilihat transaksi terbanyak ada pada bulan Januari dan terbanyak kedua ada pada bulan Maret. Pada bulan januari total pendapatannya mencapai 116291.868 USD


Pertanyaan 2: Apa metode pembayaran yang memiliki transaksi terbanyak?

In [ ]:
df['Payment'].value_counts()

Ewallet        345
Cash           344
Credit card    311
Name: Payment, dtype: int64

In [ ]:
hasil2 = df.groupby('Payment').sum()

In [ ]:
output_notebook()

pembayaran = [Payment for Payment, df in df.groupby('Payment')]

source = ColumnDataSource(data=dict(pembayaran=pembayaran, total=hasil2['Total']))

p = figure(x_range=FactorRange(*pembayaran), height=350, title="Total Penjualan berdasarkan Jenis Pembayaran",
           toolbar_location=None, tools="")

p.vbar(x='pembayaran', top='total', width=0.9, source=source, color="steelblue")

p.y_range.start = 0
p.xgrid.grid_line_color = None
p.xaxis.axis_label = "Jenis Pembayaran"
p.yaxis.axis_label = "Total dalam USD ($)"
p.xaxis.major_label_orientation = 1.2
p.yaxis.formatter = NumeralTickFormatter(format="0")
show(p)

In [ ]:
total=df.groupby(['Payment']).sum().sort_values('Payment',ascending=True).head(10)
total=total[['Total','gross income']].round(2)
total

Total  gross income
Payment                             
Cash         112206.57       5343.17
Credit card  100767.07       4798.43
Ewallet      109993.11       5237.77

In [ ]:
output_notebook()

# Group by 'Product line' and 'Payment' to count occurrences
grouped = df.groupby(['Product line', 'Payment']).size().reset_index(name='counts')

product_lines = grouped['Product line'].unique()
payments = grouped['Payment'].unique()

# Creating a nested list of tuples for FactorRange
y = [(product_line, payment) for product_line in product_lines for payment in payments]
counts = [grouped.loc[(grouped['Product line'] == product_line) & (grouped['Payment'] == payment), 'counts'].values[0] for product_line in product_lines for payment in payments]

source = ColumnDataSource(data=dict(y=y, counts=counts))

p = figure(y_range=FactorRange(*y), height=800, title='Plot Hitung untuk menunjukan jenis pembayaran pada tiap jenis produk',
           toolbar_location=None, tools="")

p.hbar(y='y', right='counts', height=0.9, source=source, line_color="white",
       fill_color=factor_cmap('y', palette=Spectral4, factors=payments, start=1, end=2))

p.x_range.start = 0
p.ygrid.grid_line_color = None
p.axis.minor_tick_line_color = None
p.outline_line_color = None

p.yaxis.axis_label = "Jenis Produk"
p.xaxis.axis_label = "Jumlah"

show(p)


In [ ]:
output_notebook()

# Group by 'Payment' and 'City' to count occurrences
grouped = df.groupby(['Payment', 'City']).size().reset_index(name='counts')

payments = grouped['Payment'].unique()
cities = grouped['City'].unique()

# Creating a nested list of tuples for FactorRange
x = [(payment, city) for payment in payments for city in cities]
counts = [grouped.loc[(grouped['Payment'] == payment) & (grouped['City'] == city), 'counts'].values[0] if (payment, city) in grouped[['Payment', 'City']].values else 0 for payment in payments for city in cities]

source = ColumnDataSource(data=dict(x=x, counts=counts))

p = figure(x_range=FactorRange(*x), height=350, title='Menunjukan metode pembayaran terbanyak diberbagai cabang',
           toolbar_location=None, tools="")

p.vbar(x='x', top='counts', width=0.9, source=source, line_color="white",
       fill_color=factor_cmap('x', palette=Spectral4, factors=cities, start=1, end=2))

p.y_range.start = 0
p.x_range.range_padding = 0.1
p.xaxis.major_label_orientation = 1
p.xaxis.group_text_font_size = '10pt'
p.xgrid.grid_line_color = None
p.axis.minor_tick_line_color = None
p.outline_line_color = None

p.xaxis.axis_label = "Metode Pembayaran"
p.yaxis.axis_label = "Jumlah"

show(p)

Metode pembayaran antara Ewallet dan Cash hanya selisih 1 orang saja, akan tetapi transaksi terbanyak dilakukan dengan pembayaran secara Cash. Pembelian Elektronik Accessories menjadi pembelian dengan jenis pembayaran Cash terbanyak. Sedangkan pembelian E Wallet terbanyak ada pada produk Fashion Accessories Pembayaran dengan metode E wallet banyak dilakukan di Yangon Pembayaran dengan metode Cash banyak dilakukan di Naypitaw Pembayaran dengan metode Credit card banyak dilakukan di Mandalay

Pertanyaan 3: Kapan waktu terbaik untuk melakukan promosi?

In [5]:
df['Time'] = pd.to_datetime(df['Time'])

In [6]:
df['Jam'] = df['Time'].dt.hour
df['Menit'] = df['Time'].dt.minute

In [ ]:
df.head()

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   
4  373-73-7910      A     Yangon        Normal    Male   

             Product line  Unit price  Quantity   Tax 5%     Total  ...  \
0       Health and beauty       74.69         7  26.1415  548.9715  ...   
1  Electronic accessories       15.28         5   3.8200   80.2200  ...   
2      Home and lifestyle       46.33         7  16.2155  340.5255  ...   
3       Health and beauty       58.22         8  23.2880  489.0480  ...   
4       Sports and travel       86.31         7  30.2085  634.3785  ...   

                 Time      Payment    cogs  gross margin percentage  \
0 2023-09-20 13:08:00      Ewallet  522.83                 4.761905   
1 2023-09-20 10:29:00         Cash   76.40                 4.761905   
2 2023-09-20 13:23:00  Credit card  324.31                 4.761905   
3 2023-09-20 20:33:00      Ewallet  465.76                 4.761905   
4 2023-09-20 10:37:00      Ewallet  604.17                 4.761905   

   gross income  Rating  Bulan  Tahun  Jam  Menit  
0       26.1415     9.1      1   2019   13      8  
1        3.8200     9.6      3   2019   10     29  
2       16.2155     7.4      3   2019   13     23  
3       23.2880     8.4      1   2019   20     33  
4       30.2085     5.3      2   2019   10     37  

[5 rows x 21 columns]

In [7]:
Waktu = df.groupby('Jam').sum()

<ipython-input-7-ddb0b01e972a>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  Waktu = df.groupby('Jam').sum()


In [ ]:
output_notebook()

# Convert each item in 'waktu' to a string representation
waktu = [str(Jam) for Jam, df in df.groupby('Jam')]

source = ColumnDataSource(data=dict(waktu=waktu, total=Waktu['Total']))

p = figure(x_range=waktu, height=350, title='Perbandingan Jumlah Transaksi Belanja di Setiap Jam',
           toolbar_location=None, tools="", x_axis_label='Jam', y_axis_label='Total dalam USD ($)')

p.vbar(x='waktu', top='total', width=0.9, source=source, color="steelblue")

p.y_range.start = 0
p.xgrid.grid_line_color = None
p.xaxis.major_label_orientation = 1.0
p.xaxis.axis_label_standoff = 12

show(p)


In [8]:
promosi=df.groupby(['Jam']).sum().sort_values('Jam',ascending=True)
promosi=promosi[['Total','gross income']].round(2)
promosi

<ipython-input-8-8541aeb634b5>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  promosi=df.groupby(['Jam']).sum().sort_values('Jam',ascending=True)


Total  gross income
Jam                        
10   31421.48       1496.26
11   30377.33       1446.54
12   26065.88       1241.23
13   34723.23       1653.49
14   30828.40       1468.02
15   31179.51       1484.74
16   25226.32       1201.25
17   24445.22       1164.06
18   26030.34       1239.54
19   39699.51       1890.45
20   22969.53       1093.79

In [ ]:
output_notebook()
steelblue_palette = ["#4682B4", "#5A8AB4", "#6C93B5", "#7E9BB5", "#91A4B6", "#A3ADB7"]

# Group by 'Jam' and 'City' to count occurrences
grouped = df.groupby(['Jam', 'City']).size().reset_index(name='counts')

jam = [str(j) for j in grouped['Jam'].unique()]  # Convert to string just in case
cities = grouped['City'].unique()

# Creating a nested list of tuples for FactorRange
x = [(j, city) for j in jam for city in cities]
counts = [grouped.loc[(grouped['Jam'] == int(j)) & (grouped['City'] == city), 'counts'].values[0] if (int(j), city) in grouped[['Jam', 'City']].values else 0 for j in jam for city in cities]

source = ColumnDataSource(data=dict(x=x, counts=counts))

p = figure(x_range=FactorRange(*x), height=550, width=1000, title='Menunjukan jam melakukan transaksi terbanyak di berbagai cabang',
           toolbar_location=None, tools="")

p.vbar(x='x', top='counts', width=0.9, source=source, line_color="white",
       fill_color=factor_cmap('x', palette=Spectral4, factors=cities, start=1, end=2))

p.y_range.start = 0
p.x_range.range_padding = 0.1
p.xaxis.major_label_orientation = 1.0
p.xaxis.group_text_font_size = '10pt'
p.xgrid.grid_line_color = None
p.axis.minor_tick_line_color = None
p.outline_line_color = None

p.xaxis.axis_label = "Jam"
p.yaxis.axis_label = "Jumlah"

show(p)

Secara keseluruhan dapat dilihat pada Jam 19.00 merupakan jam disaat transaksi terbanyak dilakukan. Berdasarkan tiap cabangnya dapat dibandingkan, di kota Mandalay jam 19.00 memang menjadi jam dengan pembeli terbanyak, di kota Yangon Jam 10.00 menjadi jam dengan pembeli terbanyak, dan di kota Naypitaw jam dengan pembeli terbanyak adalah jam 10.00

Pertanyaan 4: Apakah Member melakukan pembelian lebih banyak daripada Non Member?

In [ ]:
output_notebook()

# Calculate value counts
customer_counts = df['Customer type'].value_counts()
customer_data = pd.DataFrame({'Customer type': customer_counts.index, 'value': customer_counts.values})

# Calculate the cumulative sum for the angle data
customer_data['angle'] = customer_data['value']/customer_data['value'].sum() * 2 * pi

# Assign colors
customer_data['color'] = Spectral4[:len(customer_counts)]

p = figure(height=350, title="Pie Chart for Customer Type", toolbar_location=None,
           tools="hover", tooltips="@{Customer type}: @{value}", x_range=(-0.5, 1.0))

p.wedge(x=0, y=1, radius=0.4,
        start_angle=cumsum('angle', include_zero=True), end_angle=cumsum('angle'),
        line_color="white", fill_color='color', legend_field='Customer type', source=customer_data)

p.axis.axis_label=None
p.axis.visible=False
p.grid.grid_line_color = None

show(p)


In [ ]:
df['Customer type'].value_counts()

Member    501
Normal    499
Name: Customer type, dtype: int64

In [ ]:
output_notebook()

# Group the data by 'City' and 'Customer type'
grouped = df.groupby(['City', 'Customer type']).size().reset_index(name='counts')

cities = grouped['City'].unique()
customer_types = grouped['Customer type'].unique()

# Prepare the data for the grouped bars
x = [(city, ctype) for city in cities for ctype in customer_types]
counts = [grouped.loc[(grouped['City'] == city) & (grouped['Customer type'] == ctype), 'counts'].values[0] for city in cities for ctype in customer_types]

source = ColumnDataSource(data=dict(x=x, counts=counts))

p = figure(x_range=FactorRange(*x), height=350, title='Menunjukan tipe pembeli di berbagai cabang',
           toolbar_location=None, tools="")

p.vbar(x='x', top='counts', width=0.9, source=source, line_color="white",
       fill_color=factor_cmap('x', palette=Spectral4, factors=customer_types, start=1, end=2))

p.y_range.start = 0
p.x_range.range_padding = 0.1
p.xaxis.major_label_orientation = 1.0
p.xaxis.group_text_font_size = '10pt'
p.xgrid.grid_line_color = None
p.axis.minor_tick_line_color = None
p.outline_line_color = None

p.xaxis.axis_label = "City"
p.yaxis.axis_label = "Count"

show(p)

In [ ]:
Membership = df.groupby('Customer type').sum()

In [ ]:
output_notebook()

membership = [Customer_type for Customer_type, df in df.groupby('Customer type')]

source = ColumnDataSource(data=dict(membership=membership, total=Membership['Total']))

p = figure(x_range=membership, height=350, title='Perbandingan Pendapatan dari Membership',
           toolbar_location=None, tools="", x_axis_label='Jenis Pelanggan', y_axis_label='Total dalam USD ($)')

p.vbar(x='membership', top='total', width=0.9, source=source, color="steelblue")

p.y_range.start = 0
p.xgrid.grid_line_color = None
p.xaxis.major_label_orientation = 1.0
p.xaxis.axis_label_standoff = 12
p.yaxis.formatter = NumeralTickFormatter(format="0")

show(p)

In [ ]:
keanggotaan=df.groupby(['Customer type']).sum().sort_values('Customer type',ascending=True)
keanggotaan=keanggotaan[['Total','gross income']].round(2)
keanggotaan

Total  gross income
Customer type                         
Member         164223.44       7820.16
Normal         158743.30       7559.20

Dapat terlihat, pelanggan member dengan pelanggan normal hanya memiliki selisih sedikit, dengan hal tersebut jenis pelanggan dengan kategori member membelanjakan uangnya lebih banyak dibandingkan jenis pelanggan normal

Pertanyaan 5: Apa Produk yang paling banyak dibeli?

In [ ]:
from bokeh.io import show, output_notebook
from bokeh.models import ColumnDataSource, HoverTool, DatetimeTickFormatter
from bokeh.plotting import figure

output_notebook()

# Aggregate the quantity for each date
agg_df = df.groupby('Date').agg({'Quantity': 'sum'}).reset_index()

source = ColumnDataSource(data=dict(date=agg_df.Date, quantity=agg_df['Quantity']))

p = figure(width=1200, height=350, x_axis_type="datetime", title='Diagram Garis untuk menunjukan Penjualan perhari di Supermarket')

# Main line
p.line(x='date', y='quantity', source=source, line_width=3, color="steelblue", alpha=0.8, legend_label="Quantity")



# Add hover tool
hover = HoverTool()
hover.tooltips = [("Date", "@date{%F}"), ("Quantity", "@quantity")]
hover.formatters = {'@date': 'datetime'}
p.add_tools(hover)

# Formatting datetime axis for clarity
p.xaxis.formatter = DatetimeTickFormatter(days=["%b %d"], months=["%b %Y"], years=["%Y"])

p.y_range.start = 0
p.xaxis.axis_label = 'Date'
p.yaxis.axis_label = 'Quantity'
p.title.text_font_size = '20pt'
p.legend.location = "top_left"

show(p)


In [ ]:
Produk = df.groupby('Product line').sum()

In [ ]:
output_notebook()

produk = [Product_line for Product_line, df in df.groupby('Product line')]

source = ColumnDataSource(data=dict(produk=produk, quantity=Produk['Quantity']))

p = figure(x_range=produk, height=450, width=1000, title='Perbandingan Jumlah Pembelian dari Setiap Jenis Produk',
           toolbar_location=None, tools="", x_axis_label='Jenis Produk', y_axis_label='Jumlah Pembelian')

p.vbar(x='produk', top='quantity', width=0.9, source=source, color="steelblue")

p.y_range.start = 0
p.xgrid.grid_line_color = None

p.title.text_font_size = '15pt'
p.xaxis.axis_label_standoff = 12

show(p)

In [ ]:
output_notebook()

produk = [Product_line for Product_line, df in df.groupby('Product line')]

source = ColumnDataSource(data=dict(produk=produk, total=Produk['Total']))

p = figure(x_range=produk, height=450, width=1000, title='Perbandingan Transaksi Pembelian dari Berbagai Jenis Produk',
           toolbar_location=None, tools="", x_axis_label='Jenis Produk', y_axis_label='Transaksi Pembelian (dalam USD $)')

p.vbar(x='produk', top='total', width=0.9, source=source, color="steelblue")

p.y_range.start = 0
p.xgrid.grid_line_color = None

p.title.text_font_size = '15pt'
p.xaxis.axis_label_standoff = 12

show(p)

In [ ]:
jentrans=df.groupby(['Product line']).sum().sort_values('Product line',ascending=True)
jentrans=jentrans[['Quantity','Total', 'gross income']].round(2)
jentrans

Quantity     Total  gross income
Product line                                            
Electronic accessories       971  54337.53       2587.50
Fashion accessories          902  54305.90       2586.00
Food and beverages           952  56144.84       2673.56
Health and beauty            854  49193.74       2342.56
Home and lifestyle           911  53861.91       2564.85
Sports and travel            920  55122.83       2624.90

In [ ]:
output_notebook()

# Group by 'Product line' and 'City' to count occurrences
grouped = df.groupby(['Product line', 'City']).size().reset_index(name='counts')

product_lines = grouped['Product line'].unique()
cities = grouped['City'].unique()

# Creating a nested list of tuples for FactorRange
x = [(product, city) for product in product_lines for city in cities]
counts = [grouped.loc[(grouped['Product line'] == product) & (grouped['City'] == city), 'counts'].values[0] if (product, city) in grouped[['Product line', 'City']].values else 0 for product in product_lines for city in cities]

source = ColumnDataSource(data=dict(x=x, counts=counts))

p = figure(y_range=FactorRange(*x), height=850, title="Plot Hitung untuk menunjukan jenis pembayaran pada tiap jenis produk",
           toolbar_location=None, tools="")

p.hbar(y='x', right='counts', height=0.9, source=source, line_color="white",
       fill_color=factor_cmap('x', palette=Spectral4, factors=cities, start=1, end=2))

p.x_range.start = 0
p.ygrid.grid_line_color = None

p.xaxis.axis_label = "Count"
p.yaxis.axis_label = "Product Line"

show(p)

Di kota Naypitaw produk yang paling banyak dibeli adalah Food and Beverages dan Fashion Accessories, di kota Yangon yang terbanyak dibeli adalah Home and Lifestyle dan Electronic Accessories, sedangkan di kota Mandalay yang terbanyak adalah Fashion Accessories dan Sports and Travel

Pertanyaan 6: Kota mana yang memiliki pendapatan bersih terbanyak?

In [ ]:
output_notebook()

# Calculate value counts for 'City'
city_counts = df['City'].value_counts()
city_data = pd.DataFrame({'City': city_counts.index, 'value': city_counts.values})

# Calculate the angle data for the wedges
city_data['angle'] = city_data['value']/city_data['value'].sum() * 2 *pi

# Assign colors (you might want to pick a different palette if the number of cities is not 11)
city_data['color'] = Spectral4[:len(city_counts)]

p = figure(height=350, title="Perbandingan Jumlah Pembeli tiap Cabang", toolbar_location=None,
           tools="hover", tooltips="@City: @value", x_range=(-0.5, 1.0))

p.wedge(x=0, y=1, radius=0.4,
        start_angle=cumsum('angle', include_zero=True), end_angle=cumsum('angle'),
        line_color="white", fill_color='color', legend_field='City', source=city_data)

p.axis.axis_label=None
p.axis.visible=False
p.grid.grid_line_color = None
p.legend.location = "top_left"

show(p)

In [ ]:
kota=df.groupby(['City']).sum().sort_values('City',ascending=False)
kota=kota[['Quantity','Total', 'gross income']].round(2)
kota

Quantity      Total  gross income
City                                        
Yangon         1859  106200.37       5057.16
Naypyitaw      1831  110568.71       5265.18
Mandalay       1820  106197.67       5057.03

In [ ]:
Kota = df.groupby('City').sum()

In [ ]:
output_notebook()

kota2 = [City for City, df in df.groupby('City')]

source = ColumnDataSource(data=dict(kota2=kota2, income=Kota['gross income']))

p = figure(x_range=kota2, height=450, title='Perbandingan Pendapatan Bersih di Setiap Cabang',
           toolbar_location=None, tools="", x_axis_label='Cabang Supermarket', y_axis_label='Pendapatan Bersih (dalam USD $)')

p.vbar(x='kota2', top='income', width=0.9, source=source, color="steelblue")

p.y_range.start = 0
p.xgrid.grid_line_color = None

p.title.text_font_size = '15pt'
p.xaxis.axis_label_standoff = 12

show(p)

Berdasarkan hasil diatas dapat dilihat, walaupun Yangon memiliki jumlah pembeli dan kuantitas pembelian terbanyak. Naypitaw menjadi cabang dengan jumlah transaksi pembelian terbanyak dengan total 110568.71 USD dan total pendapatan bersihnya hingga 5265.18 USD

Kesimpulan

Berdasarkan hal yang sudah dituliskan diatas, dapat ditarik beberapa kesimpulan seperti:

1. Pada bulan januari disusul bulan maret, bisa jadi dikarenakan awal tahun baru atau awal pembukaan toko.

2. Metode pembayaran antara Ewallet dan Cash hanya selisih 1 orang saja, akan tetapi transaksi terbanyak dilakukan dengan pembayaran secara Cash. Pembelian Elektronik Accessories menjadi pembelian dengan jenis pembayaran Cash terbanyak. Sedangkan pembelian E Wallet terbanyak ada pada produk Fashion Accessories. Pembayaran dengan metode E wallet banyak dilakukan di Yangon. Pembayaran dengan metode Cash banyak dilakukan di Naypitaw. Pembayaran dengan metode Credit card banyak dilakukan di Mandalay.

3. Tiap cabang memiliki waktu terbaik yang berbeda, di kota Mandalay jam 19.00 memang menjadi jam dengan pembeli terbanyak, di kota Yangon Jam 10.00 menjadi jam dengan pembeli terbanyak, dan di kota Naypitaw jam dengan pembeli terbanyak adalah jam 10.00

4. Jumlah Member dan Non member yang berbelanja dari kisaran bulan januari sampai bulan maret cenderung sama, meskipun demikian member membelanjakan uangnya lebih banyak daripada Non member. Dengan promo-promo yang diberikan kepada member, maka supermarket haruslah menambah jumlah member agar transaksi yang diberikan juga lebih banyak.

5. Di kota Naypitaw produk yang paling banyak dibeli adalah Food and Beverages dan Fashion Accessories, di kota Yangon yang terbanyak dibeli adalah Home and Lifestyle dan Electronic Accessories, sedangkan di kota Mandalay yang terbanyak adalah Fashion Accessories dan Sports and Travel

6. Dari cabang yang ada di 3 kota, kota Naypitaw memberikan pemasukan terbanyak pada supermarket, padahal jumlah pembeli yang ada di kota Naypitaw lebih sedikit dibandingkan di 2 cabang yang lain, hal ini berarti Masyarakat di kota Naypitaw lebih sering mengeluarkan banyak uang saat berbelanja.

Dengan demikian untuk dapat memaksimalkan penjualan, Supermarket perlu untuk:

1. Mengadakan Promo pada waktu tertentu agar dapat terus meningkatkan pendapatan setiap bulannya

2. Membuat aplikasi berbelanja secara online agar pemanfaatan E wallet lebih meningkat

3. Memasang iklan pada Supermarket tersebut, baik mengiklankan diri ataupun mengajak perusahaan lain untuk beriklan di Supermarket pada jam-jam terbaik

4. Meningkatkan jumlah member.

5. Memperbanyak Variasi Produk pada bagian Health and Beauty agar dapat meningkatkan juga transaksi pada jenis Health and Beauty. Selain itu dengan data tersebut kita juga dapat melihat kebiasaan masyarakat sekitar, seperti di kota Naypitaw masyarakatnya senang berbelanja makanan di luar dan senang berbelanja Fashion Accessories, di Kota Yangon yang banyak dibeli adalah kebutuhan rumah seperti Home and Lifestyle dan Electronic Accessories, dan di kota Mandalay yang terbanyak adalah Fashion Accessories dan Sports and Travel artinya dikota ini masyarakatnya senang untuk keluar dan berpergian.

6. Melakukan promosi berupa pembelian skala besar kepada pelanggan sehingga dapat menarik pelanggan untuk membeli secara besar-besaran.
